validator/client/metrics.go

```diff
@@ -161,66 +161,110 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 	if slot >= params.BeaconConfig().SlotsPerEpoch {
 		prevEpoch = (slot / params.BeaconConfig().SlotsPerEpoch) - 1
+		if v.summaryStats.startEpoch == 0 {
+			//prevEpoch may be 0 so we store startEpoch (+1). Thus a value of 0 tells us it is not set (default)
+			v.summaryStats.startEpoch = prevEpoch + 1
+		}
 	}
 	gweiPerEth := float64(params.BeaconConfig().GweiPerEth)
 	for i, pubKey := range resp.PublicKeys {
 		pubKeyBytes := bytesutil.ToBytes48(pubKey)
 		if slot < params.BeaconConfig().SlotsPerEpoch {
 			v.prevBalance[pubKeyBytes] = params.BeaconConfig().MaxEffectiveBalance
 		}
+		if _, ok := v.startBalance[pubKeyBytes]; ok == false {
+			v.startBalance[pubKeyBytes] = resp.BalancesBeforeEpochTransition[i]
+		}
 
 		fmtKey := fmt.Sprintf("%#x", pubKey)
 		truncatedKey := fmt.Sprintf("%#x", pubKey[:8])
 		if v.prevBalance[pubKeyBytes] > 0 {
 			newBalance := float64(resp.BalancesAfterEpochTransition[i]) / gweiPerEth
 			prevBalance := float64(resp.BalancesBeforeEpochTransition[i]) / gweiPerEth
+			startBalance := float64(v.startBalance[pubKeyBytes]) / gweiPerEth
 			percentNet := (newBalance - prevBalance) / prevBalance
+			percentSinceStart := (newBalance - startBalance) / startBalance
 			log.WithFields(logrus.Fields{
-				"pubKey":               truncatedKey,
-				"epoch":                prevEpoch,
-				"correctlyVotedSource": resp.CorrectlyVotedSource[i],
-				"correctlyVotedTarget": resp.CorrectlyVotedTarget[i],
-				"correctlyVotedHead":   resp.CorrectlyVotedHead[i],
-				"inclusionSlot":        resp.InclusionSlots[i],
-				"inclusionDistance":    resp.InclusionDistances[i],
-				"oldBalance":           prevBalance,
-				"newBalance":           newBalance,
-				"percentChange":        fmt.Sprintf("%.5f%%", percentNet*100),
+				"pubKey":                  truncatedKey,
+				"epoch":                   prevEpoch,
+				"correctlyVotedSource":    resp.CorrectlyVotedSource[i],
+				"correctlyVotedTarget":    resp.CorrectlyVotedTarget[i],
+				"correctlyVotedHead":      resp.CorrectlyVotedHead[i],
+				"inclusionSlot":           resp.InclusionSlots[i],
+				"inclusionDistance":       resp.InclusionDistances[i],
+				"startBalance":            startBalance,
+				"oldBalance":              prevBalance,
+				"newBalance":              newBalance,
+				"percentChange":           fmt.Sprintf("%.5f%%", percentNet*100),
+				"percentChangeSinceStart": fmt.Sprintf("%.5f%%", percentSinceStart*100),
 			}).Info("Previous epoch voting summary")
 			if v.emitAccountMetrics {
 				ValidatorBalancesGaugeVec.WithLabelValues(fmtKey).Set(newBalance)
 			}
 		}
+		v.prevBalance[pubKeyBytes] = resp.BalancesBeforeEpochTransition[i]
+	}
+
+	v.UpdateLogAggregateStats(resp, slot)
+	return nil
+}
 
+//UpdateLogAggregateStats updates and logs the summaryStats struct of a validator using the RPC response obtained during LogValidatorGainsAndLosses
+func (v *validator) UpdateLogAggregateStats(resp *ethpb.ValidatorPerformanceResponse, slot uint64) {
```

Comment:
Can you add a test for `UpdateLogAggregateStats`? Thanks

Reply:
Added.